pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -157,7 +158,8 @@ public PersistentSubscription(PersistentTopic topic, String subscriptionName, Ma
         this.subscriptionProperties = MapUtils.isEmpty(subscriptionProperties)
                 ? Collections.emptyMap() : Collections.unmodifiableMap(subscriptionProperties);
         if (topic.getBrokerService().getPulsar().getConfig().isTransactionCoordinatorEnabled()
-                && !isEventSystemTopic(TopicName.get(topicName))) {
+                && !isEventSystemTopic(TopicName.get(topicName))
+                && !ExtensibleLoadManagerImpl.isInternalTopic(topicName)) {
```

Comment:
It would be better to have this type of logic centralized. It's not a good sign when more classes get coupled with ExtensibleLoadManagerImpl.

There have been discussions of coming up with a convention for Pulsar for system topics, but no decisions came out of that (which was disappointing). 
Trying to search for the discussion. Came across https://lists.apache.org/thread/1lndgf1hx821fc12t0pc6j6zdrhkntht and https://github.com/apache/pulsar/pull/20397 . Not that one.
It's this discussion: https://lists.apache.org/thread/oz79m0f2nw059jctq4cmms74yq5n2l1m

That's just useful background knowledge to have. In this case, I'd look for a practical solution. However, there is a need to cleanup messes when working on small changes like this one. 

In the Topic interface we already have `isSystemTopic()`:
https://github.com/apache/pulsar/blob/59782b3c1f63baa68b2f5f33a33f1b00212159d7/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Topic.java#L322
This is used in many locations, but there's also other ways to detect system topics.

We shouldn't be adding yet another way for ExtensibleLoadManagerImpl topics, but instead find a way to cleanup the mess. I hope this makes sense.

Reply:
We have also `SystemTopicNames`:
https://github.com/apache/pulsar/blob/9918bced4465e0b0746a7959550c90cb76ae945f/pulsar-common/src/main/java/org/apache/pulsar/common/naming/SystemTopicNames.java#L71-L97

As you can see, it's not consistent. The `isSystemTopic` method isn't an intersection of all the possible subtypes.
And we also have the `Topic.isSystemTopic` method which is another way and not consistent. 
This mess should be cleaned up or at least not made worse...

Reply:
We had a discussion about this and in the short term, this could be done, but in the long term, we need a proper solution for system topics. @asafm suggested a modular solution where different "plugins" would register their system topic rules to a system topic manager etc. . This sounds like a useful long term solution to the problem.

We discussed that SystemTopicNames cannot reference the ExtensibleLoadManagerImpl and that's why moving the method call there isn't possible.